### PR TITLE
fix: disable cutoff where it hurts concurrency

### PR DESCRIPTION
**Pull request by rgrinberg**
Make it possible to enable it with the "speculative_cutoff" internal
config option

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>

### CODE REVIEW

**Review by snowleopard — APPROVED** [review]
The changes look good but I'm not sure about the name: "speculative" doesn't sound like the right word here because there isn't much speculation going on. 

I'd use something longer but hopefully clearer: `cutoffs_that_reduce_concurrency_in_watch_mode`.

**Review by snowleopard — APPROVED** [review]
Thanks!

File: CHANGES.md [hunk 1]
```diff
@@ -1,6 +1,9 @@
 Unreleased
 ----------
 
+- Dune in watch mode no longer builds concurrent rules in serial (#7395
```
Comment (Alizter) [inline CHANGES.md:4]: fixes https://github.com/ocaml/dune/issues/5549 ?

Comment (rgrinberg) [reply CHANGES.md:4]: We'll claim that we fixed it once we bring back cutoff without the penalty.

**Comment by rgrinberg** [comment]
Waiting for @jchavarri to confirm the fix

**Review by jchavarri — APPROVED** [review]
Thanks @rgrinberg !

I have spent some time testing this branch in watch mode, adding changes in multiple files on both in Ahrefs internal repo and the [public repro](https://github.com/jchavarri/dune_repro_issue_5549) and concurrency stayed optimal at all times during both tests. 🎉

**Comment by snowleopard** [comment]
@jchavarri Awesome, thank you! I'll give it a try internally too once I'm back to work.

**Pull request merged**